### DISCUSSION TITLE
How can I handle this. There are 2 content-types. One is Content-Type: application/json; charset=utf-8 the other is Content-Type: audio/wav.

### DISCUSSION THREAD

**Discussion by askinucuncu** [square/retrofit#3682] [Q&A]
Hi. The following value comes from the service. How can I handle this. There are 2 content-types. One is Content-Type: application/json; charset=utf-8 the other is Content-Type: audio/wav.

--379bfc09-27b4-4a5a-a85a-e699f5129379
Content-Type: application/json; charset=utf-8
Content-Disposition: form-data; name=Json

{
   "type":"message",
   "id":"9abcba9b-5a19-4020-8ac2-1ec940857768",
   "timestamp":"2022-01-10T12:33:44.6423693+00:00",
   "textFormat":null,
   "speak":"<speak>Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book.</speak>"
}

--379bfc09-27b4-4a5a-a85a-e699f5129379
Content-Type: audio/wav
Content-Disposition: form-data; filename=AudioFromTts.wav; name=AudioFromTts
Content-Encoding: binary

RIFF4WAVEfmt �>}data�3
  
!"%,&'..5@A??B5@88999699BB69299=26+>B.299"2'6'"2''""2.

--379bfc09-27b4-4a5a-a85a-e699f5129379--

**Comment by JakeWharton** [comment]
Right now headers are not passed to converters (feature request is #2987).

Your best bet is to write a converter which uses `peek()` on the stream to see if the first few bytes match the WAV header and then conditionally pass the body to either the JSON converter or the converter for the WAV. Then you will return that in some kind of "either" type that can convey to the caller which format was parsed and the data of that format.
